Sources/_StringProcessing/Algorithms/Algorithms/Contains.swift

```diff
@@ -74,6 +74,6 @@ extension BidirectionalCollection where SubSequence == Substring {
   @_disfavoredOverload
   @available(SwiftStdlib 5.7, *)
   public func contains(_ regex: some RegexComponent) -> Bool {
-    _contains(RegexConsumer(regex))
+    (try? regex.regex.firstMatch(in: self[...])) != nil
```

Comment:
I've been trying to use the `Regex` APIs for all of these internal calls, so we don't have quite as serpentine a call stack. That said, `contains` should call `firstRange`, so that it gets any optimization we can do around ignoring capture groups.

Reply:
Are you planning on doing this in this PR or is that blocked on something?